userspace/falco/app_actions/select_event_sources.cpp

```diff
@@ -17,65 +17,54 @@ using namespace falco::app;
-		{
-			return run_result::fatal("You can not mix --enable-source and --disable-source");
-		}
+		return run_result::ok();
+	}
 
-		if (!m_options.enable_sources.empty())
+	if (!m_options.enable_sources.empty() && !m_options.disable_sources.empty())
+	{
+		return run_result::fatal("You can not mix --enable-source and --disable-source");
+	}
+
+	if (!m_options.enable_sources.empty())
+	{
+		m_state->enabled_sources.clear();
+		for(const auto &src : m_options.enable_sources)
 		{
-			m_state->enabled_sources.clear();
-			for(const auto &src : m_options.enable_sources)
+			if (m_state->loaded_sources.find(src) == m_state->loaded_sources.end())
 			{
-				if (m_state->loaded_sources.find(src) == m_state->loaded_sources.end())
-				{
-					return run_result::fatal("Attempted enabling an unknown event source: " + src);
-				}
-				m_state->enabled_sources.insert(src);
+				return run_result::fatal("Attempted enabling an unknown event source: " + src);
 			}
+			m_state->enabled_sources.insert(src);
 		}
-		else if (!m_options.disable_sources.empty())
+	}
+	else if (!m_options.disable_sources.empty())
+	{
+		for(const auto &src : m_options.disable_sources)
 		{
-			// this little hack ensure that the single-source samentic gets respected
-			// todo(jasondellaluce): remove this insert once we support multiple enabled event sources
-			m_state->enabled_sources = m_state->loaded_sources;
-
-			for(const auto &src : m_options.disable_sources)
+			if (m_state->loaded_sources.find(src) == m_state->loaded_sources.end())
 			{
-				if (m_state->loaded_sources.find(src) == m_state->loaded_sources.end())
-				{
-					return run_result::fatal("Attempted disabling an unknown event source: " + src);
-				}
-				m_state->enabled_sources.erase(src);
+				return run_result::fatal("Attempted disabling an unknown event source: " + src);
 			}
+			m_state->enabled_sources.erase(src);
 		}
+	}
 
-		if(m_state->enabled_sources.empty())
-		{
-			return run_result::fatal("Must enable at least one event source");
-		}
-
-		// these two little hacks ensure that the single-source samentic gets respected
-		// todo(jasondellaluce): remove these two once we support multiple enabled event sources
-		if(m_state->enabled_sources.size() > 1)
-		{
-			return run_result::fatal("Can not enable more than one event source");
-		}
-		if(*m_state->enabled_sources.begin() == falco_common::syscall_source)
-		{
-			m_state->inspector->m_input_plugin = nullptr;
-		}
-
-		/* Print all enabled sources. */
-		std::ostringstream os;
-		std::copy(m_state->enabled_sources.begin(), m_state->enabled_sources.end(), std::ostream_iterator<std::string>(os, ","));
-		std::string result = os.str();
-		result.pop_back();
-		falco_logger::log(LOG_INFO, "Enabled event sources: " + result + "\n");
+	if(m_state->enabled_sources.empty())
+	{
+		return run_result::fatal("Must enable at least one event source");
 	}
 
+	/* Print all enabled sources. */
```

Comment:
I'd move this to another app action :)

Reply:
I'm not sure! I think as app actions as pieces of Falco behavior that depend on its options and configurations. Here we print the list of sources unconditionally and I'm not sure it's worth isolating this code in an action just for this little purpose. WDYT?

Reply:
No problem, not a big deal anyway!